src/pages/office/components/d255/d255.ts

```diff
@@ -18,14 +24,22 @@ export class D255Component implements OnChanges {
 
   private formControl: FormControl;
 
+  constructor(private outcomeBehaviourProvider: OutcomeBehaviourMapProvider) { }
+
   ngOnChanges(): void {
     if (!this.formControl) {
-      this.formControl = new FormControl(null, [Validators.required]);
+      this.formControl = new FormControl(null);
       this.formGroup.addControl('d255', this.formControl);
     }
-    if (this.d255 !== null) {
-      this.formControl.patchValue(this.d255 ? 'Yes' : 'No');
+    const visibilityType = this.outcomeBehaviourProvider.getVisibilityType(this.outcome, 'd255');
```

Comment:
magic string

Reply:
removed magic strings and introduced enum